csxtools/utils.py

```diff
@@ -128,6 +127,93 @@ def get_fastccd_images(light_header, dark_headers=None,
     return _correct_fccd_images(events, bgnd, flat, gain)
 
 
+def get_axis_images(light_header, dark_header=None, flat=None, tag=None, roi=None):
+    """Retreive and correct AXIS1 Images from associated headers
+
+    Retrieve AXIS1 Images from databroker and correct for:
+
+    -   Bad Pixels (converted to ``np.nan``)
+    -   Backgorund.
+    -   Flatfield correction.
+    -   Rotation (returned images are rotated 90 deg cw)
+
+    Parameters
+    ----------
+    light_header : databorker header
+        This header defines the images to convert
+
+    dark_headers : databroker headers , optional
+        The header is the dark images.
+
+    flat : array_like
+        Array to use for the flatfield correction. This should be a 2D
+        array sized as the last two dimensions of the image stack.
+
+
+    tag : string
+        Data tag used to retrieve images. Used in the call to
+        ``databroker.get_images()``. If `None`, use the defualt from
+        the settings.
+
+    roi : tuple
+        coordinates of the upper-left corner and width and height of
+        the ROI: e.g., (x, y, w, h)
+
+    Returns
+    -------
+    dask.array : corrected images
+
+    """
+    flipped_image = _get_axis1_images(light_header, dark_header, flat, tag, roi)
+    return flipped_image[..., ::-1]
+
+
+def _get_axis1_images(light_header, dark_header=None, flat=None, tag=None, roi=None):
+
+    if tag is None:
+        tag = detectors["axis1"]
+
+    # Now lets sort out the ROI
```

Comment:
note, for full transparency, the `roi` argument functionality broke when data broker transitioned to `v2` and no one has ever fixed the code as it was a low priority.  For the axis detector (being more that 16x larger ) we need to fix this, but 2 things need to be addressed. 

1. add image cropping to IOC and fix the ROI start, size references so the meta data will be correct and that we know Q no matter how the axis image is cropped to read out only a specific region
2. switch to v2 (or tiled) syntax that allows up to retrieve a specified arrays from the file on disk.

Reply:
looking for advice on what do do with this roi code, as it probably re-usable given that something in databroker is what broke this functionality, the code worked as desired when it was contributed.

Reply:
the code I am referring to starts on line 177.  just looking form a second reviewer to aggree or disagree.

Reply:
leave it